pennylane/plugins/tests/test_gates.py

```diff
@@ -0,0 +1,489 @@
+    def test_two_qubit_parameters(self, device, init_state, op, func, theta, tol, skip_if):
+        """Test parametrized two qubit gates."""
+        n_wires = 2
+        dev = device(n_wires)
+        capabilities = dev.__class__.capabilities()
+        skip_if("model" not in capabilities or not capabilities["model"] == "qubit")
+
+        rnd_state = init_state(n_wires)
+
+        rnd_state = init_state(n_wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.QubitStateVector(rnd_state, wires=range(n_wires))
+            op(theta, wires=range(n_wires))
+            return qml.probs(wires=range(n_wires))
+
+        res = circuit()
+
+        expected = np.abs(func(theta) @ rnd_state) ** 2
+        assert np.allclose(res, expected, atol=tol(dev.analytic))
+
+    @pytest.mark.parametrize("mat", [U, U2])
+    def test_qubit_unitary(self, device, init_state, mat, tol, skip_if):
+        """Test QubitUnitary gate."""
+        n_wires = int(np.log2(len(mat)))
+        dev = device(n_wires)
+        capabilities = dev.__class__.capabilities()
+        skip_if("model" not in capabilities or not capabilities["model"] == "qubit")
+
+        rnd_state = init_state(n_wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.QubitStateVector(rnd_state, wires=range(n_wires))
+            qml.QubitUnitary(mat, wires=list(range(n_wires)))
+            return qml.probs(wires=range(n_wires))
+
+        res = circuit()
+
+        expected = np.abs(mat @ rnd_state) ** 2
+        assert np.allclose(res, expected, atol=tol(dev.analytic))
+
+    @pytest.mark.parametrize("op, mat", three_qubit)
+    def test_three_qubit_no_parameters(self, device, init_state, op, mat, tol, skip_if):
+        """Test three qubit gates without parameters."""
+        n_wires = 3
+        dev = device(n_wires)
+        capabilities = dev.__class__.capabilities()
+        skip_if("model" not in capabilities or not capabilities["model"] == "qubit")
+
+        rnd_state = init_state(n_wires)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.QubitStateVector(rnd_state, wires=range(n_wires))
+            op(wires=[0, 1, 2])
+            return qml.probs(wires=range(n_wires))
+
+        res = circuit()
+
+        expected = np.abs(mat @ rnd_state) ** 2
+        assert np.allclose(res, expected, atol=tol(dev.analytic))
+
+
+@flaky(max_runs=10)
+class TestInverseGatesQubit:
+    """Test the device's probability vector after application of inverse of gates."""
+
+    @pytest.mark.parametrize("op,mat", single_qubit)
+    def test_single_qubit_no_parameters(self, device, init_state, op, mat, tol, skip_if):
+        """Test inverse PauliX application."""
+        n_wires = 1
+        dev = device(n_wires)
+        capabilities = dev.__class__.capabilities()
+        skip_if("model" not in capabilities or not capabilities["model"] == "qubit")
+        skip_if("inverse_operations" not in capabilities)
```

Comment:
Should there be some logging done in such cases to inform that these tests were skipped?